src/Http/Controllers/Admin/ForgotPasswordController.php

```diff
@@ -20,23 +23,45 @@ class ForgotPasswordController extends Controller
 
     use SendsPasswordResetEmails;
 
-    public function broker()
-    {
-        return Password::broker('twill_users');
-    }
+    /**
+     * @var PasswordBrokerManager
+     */
+    protected $passwordBrokerManager;
 
-    public function showLinkRequestForm()
-    {
-        return view('twill::auth.passwords.email');
-    }
+    /**
+     * @var ViewFactory
+     */
+    protected $viewFactory;
 
     /**
      * Create a new controller instance.
      *
+     * @param Application $app
+     * @param PasswordBrokerManager $passwordBrokerManager
+     * @param ViewFactory $viewFactory
+     * @param Config $config
```

Comment:
Doc blocks can be removed because it is redundant due to type-hinted parameters and simple logic.

Reply:
They are not quite redundant. They keep us safe in case somebody changes some hint by mistake - he/she we will be able to automatically see that something has changed and it's not accepted by descendants or usages. Docblocks are also used for defining class/method descriptions, so in case at some point of time we get some kind of descriptions, it would be easier to add them. Generally speaking, docblocks are included in every package/framework of any language which is considered as well-documented, so I would leave them here and everywhere. 

@ifox @yanhao-li  What do you guys think?

Reply:
1.) Units tests will prevent us from that kind of mistake.
2.) I was thinking the same before type-hintings were not useful and diverse(void, nullables, etc.) as today. And clean code should explain itself with its name and flow. Comments are generally distracting unless the method is too complicated and couldn't be refactored more, and the code is already well-formed. Especially doc blocks are taking a lot of space.

```php
/**
 * Create a new controller instance.
 *
 * @param Application $app
 * @param PasswordBrokerManager $passwordBrokerManager
 * @param ViewFactory $viewFactory
 * @param Config $config
 * @return void
 */
public function __construct(
    Application $app,
    PasswordBrokerManager $passwordBrokerManager,
    ViewFactory $viewFactory,
    Config $config
)
```
I don't think `Create a new controller instance.` sentence nor the param types are useful here. Parameter and method descriptions are generally used in Public API's, and they are useful there.

Reply:
But we don't have any unit tests right now, right?
Ok, I will remove the ones which are not useful.
But I will still love to hear what others think. @ifox @yanhao-li